grafana/grafana
# Elasticsearch: Respect time range in ad hoc filters

(fixes https://github.com/grafana/grafana/issues/51381)

when using the ad-hoc filters with Elasticsearch, to fill in the label-values, we send a query to Elasticsearch, and we need to specify the time-range.

currently we send a hard-coded 6hour-time-range.

we want to use the currently-selected-time-range, but that information does not arrive in the API call (it's the `datasource.getTagValues` call). other datasources use the `getTimeSrv` function to get access to the time-range, we do the same here.

how to test:
- make devenv soures=elasticsearch
- create a dashboard, save it
- go to dashboard-settings, variables, add a variable of type 'ad-hoc-filter', name it `hello`, choose the elasticsearch datasource for it, press [update], save dashboard
- go back to the dashboard
- at the top-left corner you should be able to see the `hello` text with a `[+]` next to it. click it. you should be able to choose label-names and label-values there
- open the browser-devtools, look at the AJAX-requests
- everytime you open the label-values (not label-keys) select-box, an ajax request `_msearch` will run. look at the data sent from the browser, it will be a JSON-ish structure. find this part: `{"@timestamp":{"gte":"number1","lte":"number2"`. those two numbers are the time-range, in milliseconds. calculate the difference of those two numbers `number2 - number1`, it should be the same as the currently chosen time-range, in milliseconds
- change the currently selected time-range to a different value, and repeat the test, and verify that `number2 - number1` is still correct
